apps/emqx_resource/src/emqx_resource_manager.erl

```diff
@@ -229,14 +231,25 @@ set_resource_status_connecting(ResId) ->
 -spec lookup(resource_id()) -> {ok, resource_group(), resource_data()} | {error, not_found}.
 lookup(ResId) ->
     case safe_call(ResId, lookup, ?T_LOOKUP) of
-        {error, timeout} -> ets_lookup(ResId);
+        {error, timeout} -> ets_lookup(ResId, [metrics]);
         Result -> Result
     end.
 
-%% @doc Lookup the group and data of a resource
+%% @doc Lookup the group and data of a resource from the cache
 -spec ets_lookup(resource_id()) -> {ok, resource_group(), resource_data()} | {error, not_found}.
 ets_lookup(ResId) ->
+    ets_lookup(ResId, []).
+
+%% @doc Lookup the group and data of a resource from the cache
+-spec ets_lookup(resource_id(), [Option]) ->
+    {ok, resource_group(), resource_data()} | {error, not_found}
+when
+    Option :: metrics.
+ets_lookup(ResId, Options) ->
```

Comment:
Nit: this is the inherited naming, but maybe we could use something like `cache_lookup` instead of `ets_lookup` for external usage, not exposing the internals in naming.